crates/js-component-bindgen/src/transpile_bindgen.rs

```diff
@@ -1435,7 +1448,11 @@ impl<'a> Instantiator<'a, '_> {
         match def {
             CoreDef::Export(e) => self.core_export(e),
             CoreDef::Trampoline(i) => format!("trampoline{}", i.as_u32()),
-            CoreDef::InstanceFlags(i) => format!("instanceFlags{}", i.as_u32()),
+            CoreDef::InstanceFlags(i) => {
+                // SAFETY: short-lived borrow-mut.
+                self.used_instance_flags.borrow_mut().insert(*i);
```

Comment:
These are quite a few hoops just to deal with a `u32`. I'd suggest rather just cloning it into the set directly to avoid all the other borrow complexity. (it's underlying datatype is just a u32)

Reply:
oh it's because I'm mutating the `used_instance_flags` that I need the inner mutability with `RefCell`. The alternatives that I've discarded were the following:

- have `core_def` take `&mut sef`: didn't work with existing code where this would cause mutable vs immutable conflicts
- pass a `instance_flags: &mut BTreeSet` parameter
  - if that variable lived in `self` and is mutably borrowed at call sites, we get back to the previous bullet item
  - if not, then I found it a bit invasive to pass it down to all the callers, from the top-level.

Hence the inner mutability. Do you have a preference over how to achieve this?

Reply:
Ahh I see, makes sense, thanks for explaining the reasoning here!